pkg/util/testing/client.go

```diff
@@ -64,3 +66,36 @@ func (b *builderIndexer) IndexField(ctx context.Context, obj client.Object, fiel
 func AsIndexer(builder *fake.ClientBuilder) client.FieldIndexer {
 	return &builderIndexer{ClientBuilder: builder}
 }
+
+type EventRecord struct {
```

Comment:
Maybe not needed if we use the `record.FakeRecorder`?

Reply:
I've looked into `record.FakeRecorder` before doing this.